core/tests/conftest.py

```diff
@@ -1,6 +1,35 @@
+from pathlib import Path
+
 import pytest
 from typing import Callable
+import subprocess
 from testcontainers.core.container import DockerClient
+import sys
+
+PROJECT_DIR = Path(__file__).parent.parent.parent.resolve()
+
+
+def pytest_configure(config: pytest.Config) -> None:
+    """
+    Add configuration for custom pytest markers.
+    """
+    config.addinivalue_line(
+        "markers",
+        "inside_docker_check: test used to validate DinD/DooD are working as expected",
+    )
+
+
+@pytest.fixture(scope="session")
+def python_testcontainer_image() -> str:
+    """Build an image with test containers python for DinD and DooD tests"""
+    py_version = ".".join(map(str, sys.version_info[:2]))
+    image_name = f"testcontainers-python:{py_version}"
+    subprocess.run(
+        [*("docker", "build"), *("--build-arg", f"PYTHON_VERSION={py_version}"), *("-t", image_name), "."],
```

Comment:
Then is still as good as before, as the Makefile also was using sudoless docker.

Only difference people will know that there is an issue because before they probably never cared to run `make test-dind`.
So I would say dosn't matter. 
Either people will be so smart and see that if subprocess fails that it is there setup in which they required `sudo` and locally adopt the test or they will simply ignore the failing test and still create an MR that run properly in the pipeline.

Or they will create an Issue at which point we can take care of this.

Reply:
cool, let me try it on a vm :)

Reply:
Or put in other words: That only is for testing, as long as people don't run into problems and it works in the pipeline I would **not** invest time into this.
There are many more pressuring open issues inside the project.

Reply:
I'm asking because I had some unrelated issues that made me rerun all the tests and now (once I fixed the other stuff) I'm left with this, in any case I'll try and recreate all the env as sudoless.
So sadly I assume this does break :( and currently I'm effected.

Reply:
Sorry to hear.
So on your dev machince you only can run docker with `sudo docker` ?
Does is brake also in the pipeline here in gitlab?

Reply:
ok, so updating if anyone needs in the future, the workaround is very trivial just run with
```python
[*("sudo", "docker", "build"), *("--build-arg", f"PYTHON_VERSION={py_version}"), *("-t", image_name), "."],
```
**once** and you should be good :) as the image will be created and available. 

+ We should consider adding some remark in the docs about assuming all devs are using sudoless docker.

Reply:
No not *once* -> everytime you update the code.
As the code is included in the docker image.

If you already have the error in front of you, why not catch it and rerun it with sudo?
I assume there is permission error or similar?

First try without sudo and when running into the troubles, just add it.

What do you think of this idea?

Reply:
Created an issue to document this https://github.com/testcontainers/testcontainers-python/issues/749

Reply:
Just to clarify, once is just to get the tests working, if I don't change anything `dind` related I tend to trust the CI for those kind of changes, but as a safety net I'll try and move my env to use sudoless.

The correct approach in my options is to stop using `subprocess.run` for docker related activity, as we have the docker SDK :) I also noted this in the ticket for the long run.